src/app/guards/auth-guard.ts

```diff
@@ -0,0 +1,32 @@
+import { CanActivate, Router, RouterStateSnapshot, ActivatedRouteSnapshot } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { AuthService } from '../services/auth.service';
+import { Observable, of } from 'rxjs';
+import { take, map, catchError } from 'rxjs/operators';
+
+@Injectable()
+export class AuthGuard implements CanActivate{
+    constructor(private router: Router, private authService: AuthService) { }
+
+    canActivate(activatedRoute: ActivatedRouteSnapshot, routerState: RouterStateSnapshot): boolean | Observable<boolean> {
+        if (this.authService.user) {
+            return true;
+        }
+
+        // Making api call to validate token and get user if application was reloaded
+        if (localStorage.getItem("token")) {
+            return this.authService.getCurrentUser().pipe(
+                 map(user => {
+                     if (user) {
+                         return true;
+                     }
+                 }),
```

Comment:
Instead of map you should use filter in this case:

filter(Boolean)

Map is used for changing the incoming value and returning it. In your case all you want is to know if user exists